### PR TITLE
Add region.snippet, physicalLocation.contextRegion; remove result.sni…

### PR DESCRIPTION
**Pull request by EasyRhin0**
…ppet

### CODE REVIEW

File: src/Sarif.Converters/FortifyFprConverter.cs [hunk 1]
```diff
@@ -594,9 +594,10 @@ private void AddSnippetsToResults()
             {
                 string snippetId, snippetText;
                 if (_resultToSnippetIdDictionary.TryGetValue(result, out snippetId) &&
-                    _snippetIdToSnippetTextDictionary.TryGetValue(snippetId, out snippetText))
+                    _snippetIdToSnippetTextDictionary.TryGetValue(snippetId, out snippetText) &&
+                    result.Locations[0]?.PhysicalLocation?.Region != null)
```
Comment (lgolding) [inline src/Sarif.Converters/FortifyFprConverter.cs]: Shouldn't in be `result.Locations?[0]?.`?

File: src/Sarif.Viewer.VisualStudio/ViewModels/ViewModelLocator.cs [hunk 2]
```diff
@@ -69,31 +69,31 @@ private static SarifErrorListItem GetDesignTimeViewModel1()
             viewModel.Locations.Add(new Models.AnnotatedCodeLocationModel()
             {
                 FilePath = @"D:\GitHub\NuGet.Services.Metadata\src\Ng\Catalog2Dnx.cs",
-                Region = new CodeAnalysis.Sarif.Region(11, 1, 11, 2, 0, 0),
+                Region = new CodeAnalysis.Sarif.Region(11, 1, 11, 2, 0, 0, null),
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/ViewModels/ViewModelLocator.cs]: Perhaps `snippet = null` for clarity?

File: src/Sarif.Converters/FortifyFprConverter.cs (hunk 1 above)
Comment (EasyRhin0) [reply src/Sarif.Converters/FortifyFprConverter.cs]: Yep thanks

File: src/Sarif.Viewer.VisualStudio/ViewModels/ViewModelLocator.cs (hunk 2 above)
Comment (EasyRhin0) [reply src/Sarif.Viewer.VisualStudio/ViewModels/ViewModelLocator.cs]: Good call.

**Pull request merged**